### ISSUE TITLE
Animation bugs on older protocol - 7.72 confirmed but probably many more (up until animations were changed I guess)

### ISSUE THREAD

**Issue by Silba93** [OTCv8/otclientv8#206]
Tested and confirmed on unmodified latest otcv8, clean 7.72 dat/spr and unmodified nekiro 7.72 tfs
On cip client the animations are fine

Happens while idle and also while walking

2-frame animations work fine, outfits and monsters such at fire elemental
3-frame animations skip the 3rd frame entirely and only frame 1 and 2 are shown
6-frame animations do not skip any frames but animate very fast

![animationscompare](https://user-images.githubusercontent.com/13322094/169306637-54176d4b-55f7-4a19-aaa0-b7db65bc05ee.gif)

Here swapped the sprites to make this more clear
![animations](https://user-images.githubusercontent.com/13322094/169305016-f7653072-1001-4741-8f00-e57cef122b00.gif)

**Comment by Silba93** [comment]
> My bad, will be fixed with new update. Sorry about that.

Awesome thank you, I just tested this and yes all frames are now there however my rotworms are still animating way too fast. I did some more testing and here's what I found:

2-frame is a little bit slow compared to cip (bat, wasp, fire elemental etc)
3-frame is good, i can't tell the difference to cip (deathslicer, beholder)
4-frame and 6-frame animate way too fast (butterfly, slime, rotworm)

**Comment by Oen44** [comment]
Frame duration in Object Builder are correct?

**Comment by Silba93** [comment]
> Frames duration in Object Builder are correct?

Unmodified spr/dat so I'm going to assume yes?
I don't see reference to frame duration in object builder. I believe that is a feature when compiled with "enhanced animations".

**Comment by Oen44** [comment]
So is there like a formula for frame duration on older protocols? Anything based on total number of frames? I have no knowledge about old Tibia so you will have to help me with all the information.

**Comment by Silba93** [comment]
I'm just saying what I see, got no technical knowledge at all about this. Sorry >.<

**Comment by Silba93** [comment]
> So is there like a formula for frame duration on older protocols? Anything based on total number of frames? I have no knowledge about old Tibia so you will have to help me with all the information.

I think I've 'solved' it. Finally got the dev version to work so I could mess around :D

Regarding frame duration, I noticed all creature animations have the same "ticks per frame" on cip client so a 2-frame bat would complete 3 rounds of animations in the same time it takes a 6-frame rotworm to complete once.

In OTCV8 the "ticks per frame" is in `outfit.cpp` line 98 `ticksPerFrame = 1000 / phases` which explains why:

Bats are slow (2-frame, 500ms ticksPerFrame)
Beholders are ok (3-frame, 333ms ticksPerFrame) - **cip client appears to be somewhere around this 300ms range**
Rotworms are fast (6-frame, 166ms ticksPerFrame)

I have compiled and used `ticksPerFrame = 330` and it's very close but not perfect, I have tried anywhere between 300-350 and nothing seems perfect. However close enough(330) is probably acceptable.

I don't know how to implement this properly but hopefully I have enough information here so a proper fix can be implemented.

Comparisons:
LEFT SIDE IS CIP
RIGHT SIDE IS OTCV8 `ticksPerFrame = 330`
![almostlol](https://user-images.githubusercontent.com/13322094/171252656-f7b1f539-5349-489a-9e65-abd70161fac7.gif)


LEFT SIDE IS CIP
RIGHT SIDE IS VANILLA OTCV8 `ticksPerFrame = 1000 / phases`
![original](https://user-images.githubusercontent.com/13322094/171252737-f9ca1429-23af-40ca-9c8a-6e0955222c41.gif)
